models/clm/CAM_bias_correction/plot_diagnostic.m

```diff
@@ -0,0 +1,641 @@
+                '-', 'color',rgb('darkred'), 'linewidth', mean_width); hold on
+         case 'CAM4'
+         j=plot(repelem(mean(Sa_tbot,1),1,6), ...
+                '-', 'color',rgb('darkred'), 'linewidth', mean_width); hold on
+       end
+       k=plot(ta_1hr, '-', 'color',rgb('black'),'linewidth', mean_width);
+       grid on
+       legend([j,k], [n,' Ensemble Mean'], 'Tower Met','FontSize',fontpt);
+
+       ylabel('Temperature (K)','FontSize', fontpt)
+
+       axes(ha(3));
+       j=plot(repelem(mean(Faxa_rainl,1),1,3)'+ ...
+              repelem(mean(Faxa_rainc,1),1,3)'+ ...
+              repelem(mean(Faxa_snowl,1),1,3)'+ ...
+              repelem(mean(Faxa_snowc,1),1,3)', ...
+              '-', 'color',rgb('darkred'), 'linewidth', mean_width); hold on
+       k=plot(ppt_1hr, '-', 'color',rgb('black'),'linewidth', mean_width);
+       grid on
+       legend([j,k], [n,' Ensemble Mean'], 'Tower Met','FontSize',fontpt);
+
+       ylabel('Total Precipitation (mm s^{-1})','FontSize', fontpt)
+
+       linkaxes(ha(1:3), 'x');
+
+
+       figure(3)
+       scrsz = get(0,'ScreenSize');
+       set(gcf,'Position',[1 1 scrsz(3) scrsz(4)]);
+       fontpt=12;
+       mean_width=1.5;
+       member_width=0.5;
+       window=10;
+
+       ha=tight_subplot(2,1, [.02,.08],[.10,.06],[.08,.06]);
+
+       axes(ha(1));
+       switch n
+         case 'CAM6'
+         j=plot(repelem(mean(Sa_pbot,1),1,3), ...
+                '-', 'color',rgb('darkred'), 'linewidth', mean_width); hold on
+         case 'CAM4'
+         j=plot(repelem(mean(Sa_pbot,1),1,6), ...
+                '-', 'color',rgb('darkred'), 'linewidth', mean_width); hold on
+       end
+       k=plot(ps_1hr, '-', 'color',rgb('black'),'linewidth', mean_width);
+       grid on
+       legend([j,k], [n,' Ensemble Mean'], 'Tower Met','FontSize',fontpt);
+
+       ylabel('Pressure (Pa)','FontSize', fontpt)
+       title(['Niwot Ridge Meteorology ', yearstr] , 'FontSize',fontpt);
+
+       axes(ha(2));
+       switch n
+         case 'CAM6'
+         j=plot(repelem(mean(Faxa_lwdn,1),1,3), ...
+                '-', 'color',rgb('darkred'), 'linewidth', mean_width); hold on
+         case 'CAM4'
+         j=plot(repelem(mean(Faxa_lwdn,1),1,6), ...
+                '-', 'color',rgb('darkred'), 'linewidth', mean_width); hold on
+       end
+       j=plot(repelem(mean(Faxa_lwdn,1),1,3), ...
+              '-', 'color',rgb('darkred'), 'linewidth', mean_width); hold on
+       k=plot(lw_1hr, '-', 'color',rgb('black'),'linewidth', mean_width);
+       grid on
+       legend([j,k], [n,' Ensemble Mean'], 'Tower Met','FontSize',fontpt);
+
+       ylabel('LW radiation (W m^{-1})','FontSize', fontpt)
+       linkaxes(ha(1:2), 'x');
+
+display('  ')
+display(['Finished tower ',n,' diagnostic plots. Press enter to proceed to adjusted ',n,' plots..'])
+display('  ')
+pause
+close all
+
+
+     % For plotting purposes only, make these 3 hour products into 1 hour products for easier comparison 
```

Comment:
The rest of the script seems to have a lot of redundancy, which could (hopefully) benefit from defining functions.   There might be opportunities above here too,
but I haven't looked in detail at how that would be done.